consumer.go

```diff
@@ -606,14 +617,60 @@ func (child *partitionConsumer) parseResponse(response *FetchResponse) ([]*Consu
 
 			messages = append(messages, messageSetMessages...)
 		case defaultRecords:
+			for _, abortedTransaction := range abortedTransactions {
+				if abortedTransaction.FirstOffset > records.RecordBatch.LastOffset() {
+					break
+				}
+				// add aborted transaction to abortedProducer list and depile abortedTransactions
+				abortedProducerIDs[abortedTransaction.ProducerID] = none{}
+				abortedTransactions = abortedTransactions[1:]
```

Comment:
https://play.golang.org/p/7C8k2ijtvWX

What are you expecting to happen after line 626, it seems that loop still iterate over original slice?

Reply:
it is expected that loop l620 iterate over original slice.
Poping the slice is used for the loop l610

Basically how it works:
- Consumer iterate over each batches (l610)
- For each batch, add to the list on aborted producer every transaction that started in a previous offset AND is not yet resolved. (l620)
- Whenever we encounter an abortedTransaction control record: resolve this transaction.
- For a given producer, transactions cannot cross each others.

The whole point of unstacking `abortedTransactions` is just to make sure we add them only once to the list of current (as in for current offset) aborted transactions and never again. 
Even more so  after transaction is resolved and the same producer may have actually  produced a committed batch displayed in the same fetchResponse.

Reply:
comment in code have been updated. I hope it is clear enough now